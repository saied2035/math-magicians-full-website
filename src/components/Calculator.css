#calculator {
  width: 60vw;
  margin: 4vh auto;
}

small {
  text-align: center;
  margin-bottom: 2px;
  color: red;
  font-size: 17px;
  font-weight: bold;
}

#result-section {
  background: #858694;
  color: #fff;
  font-size: 40px;
  padding: 2rem;
}

#result {
  display: block;
  text-align: right;
}

#opertations-section {
  padding: 0;
  margin: 0;
  list-style: none;
  font-size: 30px;
}

.row {
  width: 100%;
  border-top: 1.5px solid #bababa;
  border-bottom: 2px solid #bababa;
}

button {
  border: none;
}

.cell {
  font-size: 30px;
  text-align: center;
  display: inline-block;
  background: #e0e0e0;
  padding: 1.5rem 0;
  width: 15vw;
  box-sizing: border-box;
  border-right: 1px solid #bababa;
}

.cell:last-child {
  background: #f5913e;
  border-right: none;
}

.row:last-child > .cell:nth-last-child(3) {
  width: 30vw;
}

span {
  pointer-events: auto;
  cursor: default;
}
