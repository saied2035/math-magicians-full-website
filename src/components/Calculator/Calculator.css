#calculator {
  width: 40vw;
  margin: 4vh 2% 0 auto;
}

small {
  text-align: center;
  margin-bottom: 2px;
  color: red;
  font-size: 17px;
  font-weight: bold;
}

#result-section {
  background: #1B2021;
  color: #E3DCC2;
  font-weight: 600;
  font-size: 40px;
  padding: 2rem;
}

#result {
  display: block;
  text-align: right;
}

#opertations-section {
  padding: 0;
  margin: 0;
  list-style: none;
  font-size: 30px;
}

.row {
  width: 100%;
  border-bottom: 1px solid #561d25;
}

button {
  border: none;
}

.cell {
  font-size: 20px;
  text-align: center;
  display: inline-block;
  background: #A6A867;
  color: #1B2021;
  font-weight: 600;
  padding: 1.5rem 0;
  width: 10vw;
  box-sizing: border-box;
  border-right: 1px solid #561d25;
}

.cell:last-child {
  background: #E3DC95;
  border-right: none;
}

.row:last-child > .cell:nth-last-child(3) {
  width: 20vw;
}

span {
  pointer-events: auto;
  cursor: default;
}
