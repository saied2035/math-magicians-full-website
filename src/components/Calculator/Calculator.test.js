import {
  render, screen, fireEvent,
} from '@testing-library/react';
import Calculator from './Calculator';

const symbols = [
  [
    'AC',
    '+/-',
    '%',
    String.fromCharCode(0x00F7),
  ],

  [7, 8, 9, 'x'],

  [4, 5, 6, String.fromCharCode(0x002D)],

  [1, 2, 3, String.fromCharCode(0x002B)],

  [0, '.', String.fromCharCode(0x003D)],

];

let whatToCalculate = {
  total: null,
  next: null,
  operation: null,
};

const handleCalculations = (update) => {
  whatToCalculate = { ...whatToCalculate, ...update };
};

const renderComponent = () => {
  const { container } = render(<Calculator
    symbols={symbols}
    whatToCalculate={whatToCalculate}
    calculationsHandler={handleCalculations}
  />);
  return container;
};

const updateComponent = (whatToCalculate, container) => {
  const newResult = whatToCalculate;
  render(<Calculator
    symbols={symbols}
    whatToCalculate={newResult}
    calculationsHandler={handleCalculations}
  />, { container });
};

describe('snapshot', () => {
  const calculator = render(<Calculator
    symbols={symbols}
    whatToCalculate={whatToCalculate}
    calculationsHandler={handleCalculations}
  />);

  it('take snapshot', () => {
    expect(calculator).toMatchSnapshot();
    expect(render(<Calculator
    symbols={[]}
    whatToCalculate={whatToCalculate}
    calculationsHandler={handleCalculations}
  />)).toMatchSnapshot()
  });
});

describe('sum', () => {
  it('add two numbers 1', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('8'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('+'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('4'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math')).toHaveTextContent('12');
  });

  it('add two numbers  2 ', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('8'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('+'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('1'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('0'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math')).toHaveTextContent('18');
  });

  it('multiple sum', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('8'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('+'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('5'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('+'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('6'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('+'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('9'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math')).toHaveTextContent('28');
  });
});

describe('subtraction', () => {
  it('subtract two numbers 1', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('8'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('-'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('2'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math')).toHaveTextContent('6');
  });

  it('subtract two numbers  2 ', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('2'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('0'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('-'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('8'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math')).toHaveTextContent('12');
  });

  it('multiple subtract', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('2'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('0'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('-'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('5'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('-'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('4'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('-'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('9'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math')).toHaveTextContent('2');
  });
});

describe('maltiplcation', () => {
  it('maltiplcation two numbers 1', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('8'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('x'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('2'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math')).toHaveTextContent('16');
  });

  it('multiplcation two numbers  2 ', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('1'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('0'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('x'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('9'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math')).toHaveTextContent('90');
  });

  it('multiple multiplcation', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('2'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('x'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('5'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('x'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('4'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('x'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('9'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math')).toHaveTextContent('360');
  });
});

describe('division', () => {
  it('divide two numbers 1', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('8'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('÷'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('2'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math')).toHaveTextContent('4');
  });

  it('divide two numbers  2 ', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('2'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('0'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('÷'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('4'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math')).toHaveTextContent('5');
  });

  it('multiple division', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('8'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('6'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('0'));

    updateComponent(whatToCalculate, container);    

    fireEvent.click(screen.getByText('÷'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('5'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('÷'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('2'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('÷'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('4'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math').textContent).toBe('21.5');
  });
});

describe('multiple operations', () => {
  it('multiple operation', () => {
    const container = renderComponent();

    fireEvent.click(screen.getByText('8'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('0'));

    updateComponent(whatToCalculate, container);    

    fireEvent.click(screen.getByText('+'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('2'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('-'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('3'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('0'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('x'));
      
    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('4'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('÷'));

    updateComponent(whatToCalculate, container);    

    fireEvent.click(screen.getByText('1'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('0'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('0'));

    updateComponent(whatToCalculate, container);    

    fireEvent.click(screen.getByText('+'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('0'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('.'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('9'));
   
    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('2'));
    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('+'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('4'));

    updateComponent(whatToCalculate, container);

    fireEvent.click(screen.getByText('='));

    expect(screen.getByRole('math').textContent).toBe('7');
  });
});