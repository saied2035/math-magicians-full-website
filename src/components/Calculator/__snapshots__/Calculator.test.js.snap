// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot  and AC button take snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="ma0 tc mt5 dib w-40 f2"
      >
        Let's do some math.
      </h2>
      <div
        id="calculator"
      >
        <small
          class="f5"
          role="note"
        />
        <section
          id="result-section"
        >
          <span
            id="result"
            role="math"
          >
            0
          </span>
        </section>
        <ul
          id="opertations-section"
        >
          <li
            class="row"
          >
            <button
              class="cell"
              type="button"
              value="AC"
            >
              AC
            </button>
            <button
              class="cell"
              type="button"
              value="+/-"
            >
              +/-
            </button>
            <button
              class="cell"
              type="button"
              value="%"
            >
              %
            </button>
            <button
              class="cell"
              type="button"
              value="รท"
            >
              รท
            </button>
          </li>
          <li
            class="row"
          >
            <button
              class="cell"
              type="button"
              value="7"
            >
              7
            </button>
            <button
              class="cell"
              type="button"
              value="8"
            >
              8
            </button>
            <button
              class="cell"
              type="button"
              value="9"
            >
              9
            </button>
            <button
              class="cell"
              type="button"
              value="x"
            >
              x
            </button>
          </li>
          <li
            class="row"
          >
            <button
              class="cell"
              type="button"
              value="4"
            >
              4
            </button>
            <button
              class="cell"
              type="button"
              value="5"
            >
              5
            </button>
            <button
              class="cell"
              type="button"
              value="6"
            >
              6
            </button>
            <button
              class="cell"
              type="button"
              value="-"
            >
              -
            </button>
          </li>
          <li
            class="row"
          >
            <button
              class="cell"
              type="button"
              value="1"
            >
              1
            </button>
            <button
              class="cell"
              type="button"
              value="2"
            >
              2
            </button>
            <button
              class="cell"
              type="button"
              value="3"
            >
              3
            </button>
            <button
              class="cell"
              type="button"
              value="+"
            >
              +
            </button>
          </li>
          <li
            class="row"
          >
            <button
              class="cell"
              type="button"
              value="0"
            >
              0
            </button>
            <button
              class="cell"
              type="button"
              value="."
            >
              .
            </button>
            <button
              class="cell"
              type="button"
              value="="
            >
              =
            </button>
            
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <h2
      class="ma0 tc mt5 dib w-40 f2"
    >
      Let's do some math.
    </h2>
    <div
      id="calculator"
    >
      <small
        class="f5"
        role="note"
      />
      <section
        id="result-section"
      >
        <span
          id="result"
          role="math"
        >
          0
        </span>
      </section>
      <ul
        id="opertations-section"
      >
        <li
          class="row"
        >
          <button
            class="cell"
            type="button"
            value="AC"
          >
            AC
          </button>
          <button
            class="cell"
            type="button"
            value="+/-"
          >
            +/-
          </button>
          <button
            class="cell"
            type="button"
            value="%"
          >
            %
          </button>
          <button
            class="cell"
            type="button"
            value="รท"
          >
            รท
          </button>
        </li>
        <li
          class="row"
        >
          <button
            class="cell"
            type="button"
            value="7"
          >
            7
          </button>
          <button
            class="cell"
            type="button"
            value="8"
          >
            8
          </button>
          <button
            class="cell"
            type="button"
            value="9"
          >
            9
          </button>
          <button
            class="cell"
            type="button"
            value="x"
          >
            x
          </button>
        </li>
        <li
          class="row"
        >
          <button
            class="cell"
            type="button"
            value="4"
          >
            4
          </button>
          <button
            class="cell"
            type="button"
            value="5"
          >
            5
          </button>
          <button
            class="cell"
            type="button"
            value="6"
          >
            6
          </button>
          <button
            class="cell"
            type="button"
            value="-"
          >
            -
          </button>
        </li>
        <li
          class="row"
        >
          <button
            class="cell"
            type="button"
            value="1"
          >
            1
          </button>
          <button
            class="cell"
            type="button"
            value="2"
          >
            2
          </button>
          <button
            class="cell"
            type="button"
            value="3"
          >
            3
          </button>
          <button
            class="cell"
            type="button"
            value="+"
          >
            +
          </button>
        </li>
        <li
          class="row"
        >
          <button
            class="cell"
            type="button"
            value="0"
          >
            0
          </button>
          <button
            class="cell"
            type="button"
            value="."
          >
            .
          </button>
          <button
            class="cell"
            type="button"
            value="="
          >
            =
          </button>
          
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
